editor/src/clj/util/coll.clj

```diff
@@ -43,3 +43,40 @@
       (if (and s (< i n))
         (recur (inc i) (next s))
         i))))
+
+(defn resize
+  "Returns a collection of the desired size, retaining elements from coll up to
+  the desired size. If coll does not have enough elements, fill the subsequent
+  slots in the output collection with the supplied fill-value."
+  [coll ^long new-count fill-value]
+  {:pre [(sequential? coll)
+         (not (neg? new-count))]}
+  (let [old-count (count coll)]
+    (cond
+      (< new-count old-count)
+      (into (empty coll)
+            (take new-count)
+            coll)
+
+      (> new-count old-count)
+      (into coll
+            (repeat (- new-count old-count)
+                    fill-value))
+
+      :else
+      coll)))
+
+(defn separate-by
+  "Separates items in the supplied collection into two based on a predicate.
+  Returns a pair of [true-items, false-items]. The resulting collections will
+  have the same type as the input collection."
+  [pred coll]
+  (let [empty-coll (empty coll)]
+    (reduce (fn [result item]
+              (if (pred item)
+                (pair (conj (key result) item)
+                      (val result))
+                (pair (key result)
+                      (conj (val result) item))))
+            (pair empty-coll empty-coll)
+            coll)))
```

Comment:
Unfortunately, `group-into` is currently located in another module that has a lot of extra dependencies. I don't feel like adding a dependency to `internal.util` or moving `group-into` into `util.coll` here since it would add a lot of unrelated changes to the PR.

Also, using `group-into`, would create needless garbage and overhead for the outer map containing the groups.

I do think it is important that `separate-by` makes use of transient collections, though, so I'll give that a go!